propsigner/client.go

```diff
@@ -0,0 +1,66 @@
+package propsigner
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/filecoin-project/go-state-types/crypto"
+	"github.com/filecoin-project/specs-actors/actors/builtin/market"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+	pb "github.com/textileio/go-auctions-client/gen/wallet"
+)
+
+const (
+	maxResponseMessageSize = 100 << 10
+)
+
+// RequestSignatureV1 request a signature for a deal proposal to a remote wallet.
+func RequestSignatureV1(
+	ctx context.Context,
+	h host.Host,
+	authToken string,
+	proposal market.DealProposal,
+	remoteWallet peer.ID) (*crypto.Signature, error) {
+	proposalCborBytes := &bytes.Buffer{}
+	if err := proposal.MarshalCBOR(proposalCborBytes); err != nil {
+		return nil, fmt.Errorf("marshaling deal proposal to cbor: %s", err)
+	}
+
+	req := pb.ProposalSigningRequest{
```

Comment:
Now is a good moment to explain how the request/response works using libp2p streams; shouldn't be surprising.

Basically, the client first sends a marshaled `pb.ProposalSigningRequest` to the remote wallet, which replies with a `pb.proposalSigningResponse`.